ingestion/src/metadata/utils/helpers.py

```diff
@@ -212,3 +214,30 @@ def get_raw_extract_iter(alchemy_helper) -> Iterable[Dict[str, Any]]:
     rows = alchemy_helper.execute_query()
     for row in rows:
         yield row
+
+
+def create_credential_temp_file(credentials: dict) -> str:
+    with tempfile.NamedTemporaryFile(delete=False) as fp:
+        cred_json = json.dumps(credentials, indent=4, separators=(",", ": "))
+        fp.write(cred_json.encode())
+        return fp.name
+
+
+def store_gcs_credentials(options) -> bool:
+    if not os.environ.get("GOOGLE_APPLICATION_CREDENTIALS"):
+        if options.get("credentials_path"):
+            os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = options["credentials_path"]
+            del options["credentials_path"]
```

Comment:
I don't really understand the logic here 🤔 Could you please elaborate? We might be able to find a simpler way

Reply:
We are allowing 2 options for users to enter their gcs credentials.
1. Entering the credentials directly in the json config file
2. Entering the path of the credentials file stored in local system.

This logic handles the above use cases and stores the gcs credentials in an environment variable required for gcs connection

Reply:
Uh thanks for clarifying. Could you please add that as a docstring for the function? Thanks!

Reply:
This has been moved to `credentials.py`, where we are using the GCSCredentials model

Reply:
The GCSCredentials model is used here [https://github.com/open-metadata/OpenMetadata/pull/3736/files#diff-fe842e0cd525e99a2caf7eb66994e43dd09ade131f59b6ecdd12db67b0655fd9R141](url)

and it is converted to required format for gcs afterwards